Source/Core/VideoCommon/ShaderGenCommon.h

```diff
@@ -62,6 +59,9 @@ class ShaderGeneratorInterface
 	 */
 	template<class uid_data>
 	uid_data* GetUidData() { return nullptr; }
+
+protected:
+	std::string m_buffer;
```

Comment:
The reservation is done in ShaderCode, as ShaderUid doesn't use writing, so it doesn't make sense to pre-allocate for that.